components/tools/OmeroWeb/omeroweb/webadmin/views.py

```diff
@@ -588,17 +588,15 @@ def manage_experimenter(request, action, eid=None, conn=None, **kwargs):
 
                 # Update 'AdminPrivilege' config roles for user
                 privileges = conn.get_privileges_from_form(form)
-                if privileges is None:
-                    privileges = []
                 # Only process privileges that we have permission to set
                 to_add = []
                 to_remove = []
-                for p in conn.getCurrentAdminPrivileges():
-                    if p in privileges:
-                        to_add.append(p)
-                    else:
-                        to_remove.append(p)
-
+                if privileges is not None:
+                    for p in conn.getCurrentAdminPrivileges():
+                        if p in privileges:
+                            to_add.append(p)
+                        else:
+                            to_remove.append(p)
                 conn.updateAdminPrivileges(experimenter.id, to_add, to_remove)
```

Comment:
'conn.updateAdminPrivileges(experimenter.id, to_add, to_remove)' should probably exist within the 'if' block